src/libYARP_OS/src/ThreadImpl.cpp

```diff
@@ -126,7 +126,10 @@ ThreadImpl::ThreadImpl() :
         needJoin(false),
         delegate(YARP_NULLPTR),
         synchro(0),
-        initWasSuccessful(false)
+        initWasSuccessful(false),
+        id(-1),
+        hid(YARP_NULLPTR),
```

Comment:
This is not correct, `hid` is not a pointer, and therefore it breaks the build on linux (I wonder why windows does not complain):
`hid` is a `Platform_hthread_t` that is defined (depending on the platform and build flags) as `std::thread`, `ACE_hthread_t`, or `pthread_t`
I'm not sure what is the correct way to initialize these types, perhaps just remove it from the initialization 
```c++
#if defined(YARP_HAS_CXX11)
        hid(std::thread()),
#else
        hid(0),
#endif
```